### ISSUE TITLE
Consider not replacing CUDA-supported PyTorch with CPU-only PyTorch

### ISSUE THREAD

**Issue by Ark-kun** [d8ahazard/sd_dreambooth_extension#1056]
### Kindly read the entire form below and fill it out with the requested information.

**Please find the following lines in the console and paste them below. _If you do not provide this information, your
issue will be automatically closed._**

```
Python 3.10.2 (tags/v3.10.2:a58ebcc, Jan 17 2022, 14:12:15) [MSC v.1929 64 bit (AMD64)]
Commit hash: b186d44dcd0df9d127a663b297334a5bd8258b58
Installing torch and torchvision
Installing gfpgan
Installing clip
Installing open_clip
Installing requirements for CodeFormer
Installing requirements for Web UI
Installing Deforum requirement: numexpr
Installing Deforum requirement: av
Installing Deforum requirement: pims
Installing Deforum requirement: imageio_ffmpeg
Installing Deforum requirement: rich

Installing requirements for scikit_learn

No module 'xformers'. Proceeding without it.


#######################################################################################################
Initializing Dreambooth
If submitting an issue on github, please provide the below text for debugging purposes:

Python revision: 3.10.2 (tags/v3.10.2:a58ebcc, Jan 17 2022, 14:12:15) [MSC v.1929 64 bit (AMD64)]
Dreambooth revision: bff61c6b92b79ece2e140ee240e0628dd9a4ebef
SD-WebUI revision: b186d44dcd0df9d127a663b297334a5bd8258b58

Checking Dreambooth requirements...
Ignoring tensorflow-macos: markers 'sys_platform == "darwin" and platform_machine == "arm64"' don't match your environment
Ignoring mediapipe-silicon: markers 'sys_platform == "darwin"' don't match your environment
Collecting accelerate==0.16.0
  Using cached accelerate-0.16.0-py3-none-any.whl (199 kB)
Collecting albumentations~=1.3.0
  Using cached albumentations-1.3.0-py3-none-any.whl (123 kB)
Collecting bitsandbytes==0.35.4
  Using cached bitsandbytes-0.35.4-py3-none-any.whl (62.5 MB)
Collecting diffusers==0.13.1
  Using cached diffusers-0.13.1-py3-none-any.whl (716 kB)
Collecting gitpython~=3.1.31
  Using cached GitPython-3.1.31-py3-none-any.whl (184 kB)
Collecting modelcards~=0.1.6
  Using cached modelcards-0.1.6-py3-none-any.whl (12 kB)
Collecting tensorflow==2.11.0
  Using cached tensorflow-2.11.0-cp310-cp310-win_amd64.whl (1.9 kB)
Collecting mediapipe
  Using cached mediapipe-0.9.1.0-cp310-cp310-win_amd64.whl (49.8 MB)
Collecting tqdm~=4.64.1
  Using cached tqdm-4.64.1-py2.py3-none-any.whl (78 kB)
Collecting transformers~=4.26.1
  Using cached transformers-4.26.1-py3-none-any.whl (6.3 MB)
Collecting discord-webhook~=1.1.0
  Using cached discord_webhook-1.1.0-py3-none-any.whl (12 kB)
Collecting lion-pytorch~=0.0.7
  Using cached lion_pytorch-0.0.7-py3-none-any.whl (4.3 kB)
Collecting xformers==0.0.17.dev464
  Using cached xformers-0.0.17.dev464-cp310-cp310-win_amd64.whl (112.3 MB)
Collecting importlib-metadata
  Using cached importlib_metadata-6.0.0-py3-none-any.whl (21 kB)
Collecting tensorflow-intel==2.11.0
  Using cached tensorflow_intel-2.11.0-cp310-cp310-win_amd64.whl (266.3 MB)
Collecting torch>=1.4.0
  Using cached torch-1.13.1-cp310-cp310-win_amd64.whl (162.6 MB)
Collecting pyre-extensions==0.0.23
  Using cached pyre_extensions-0.0.23-py3-none-any.whl (11 kB)
Collecting typing-inspect
  Using cached typing_inspect-0.8.0-py3-none-any.whl (8.7 kB)
Collecting tensorflow-estimator<2.12,>=2.11.0
  Using cached tensorflow_estimator-2.11.0-py2.py3-none-any.whl (439 kB)
Collecting tensorflow-io-gcs-filesystem>=0.23.1
  Using cached tensorflow_io_gcs_filesystem-0.31.0-cp310-cp310-win_amd64.whl (1.5 MB)
Collecting keras<2.12,>=2.11.0
  Using cached keras-2.11.0-py2.py3-none-any.whl (1.7 MB)
Collecting tensorboard
  Using cached tensorboard-2.11.2-py3-none-any.whl (6.0 MB)
Collecting libclang>=13.0.0
  Using cached libclang-15.0.6.1-py2.py3-none-win_amd64.whl (23.2 MB)
Collecting astunparse>=1.6.0
  Using cached astunparse-1.6.3-py2.py3-none-any.whl (12 kB)
Collecting protobuf<3.20,>=3.9.2
  Using cached protobuf-3.19.6-cp310-cp310-win_amd64.whl (895 kB)
Collecting opt-einsum>=2.3.2
  Using cached opt_einsum-3.3.0-py3-none-any.whl (65 kB)
Collecting flatbuffers>=2.0
  Using cached flatbuffers-23.3.3-py2.py3-none-any.whl (26 kB)
Collecting termcolor>=1.1.0
  Using cached termcolor-2.2.0-py3-none-any.whl (6.6 kB)
Collecting google-pasta>=0.1.1
  Using cached google_pasta-0.2.0-py3-none-any.whl (57 kB)
Collecting gast<=0.4.0,>=0.2.1
  Using cached gast-0.4.0-py3-none-any.whl (9.8 kB)
Collecting h5py>=2.9.0
  Using cached h5py-3.8.0-cp310-cp310-win_amd64.whl (2.6 MB)
Collecting wrapt>=1.11.0
  Using cached wrapt-1.15.0-cp310-cp310-win_amd64.whl (36 kB)
Collecting opencv-python-headless>=4.1.1
  Using cached opencv_python_headless-4.7.0.72-cp37-abi3-win_amd64.whl (38.1 MB)
Collecting qudida>=0.0.4
  Using cached qudida-0.0.4-py3-none-any.whl (3.5 kB)
Collecting tensorboard-data-server<0.7.0,>=0.6.0
  Using cached tensorboard_data_server-0.6.1-py3-none-any.whl (2.4 kB)
Collecting opencv-contrib-python
  Using cached opencv_contrib_python-4.7.0.72-cp37-abi3-win_amd64.whl (44.9 MB)
Collecting requests
  Using cached requests-2.28.2-py3-none-any.whl (62 kB)
Collecting zipp>=0.5
  Using cached zipp-3.15.0-py3-none-any.whl (6.8 kB)
Collecting mypy-extensions>=0.3.0
  Using cached mypy_extensions-1.0.0-py3-none-any.whl (4.7 kB)
Installing collected packages: requests, tensorboard-data-server, protobuf, mypy-extensions, zipp, wrapt, typing-inspect, tqdm, termcolor, tensorflow-io-gcs-filesystem, tensorflow-estimator, tensorboard, opt-einsum, opencv-python-headless, libclang, keras, h5py, google-pasta, gast, flatbuffers, astunparse, torch, tensorflow-intel, qudida, pyre-extensions, opencv-contrib-python, importlib-metadata, xformers, transformers, tensorflow, modelcards, mediapipe, lion-pytorch, gitpython, discord-webhook, diffusers, bitsandbytes, albumentations, accelerate
  Attempting uninstall: requests
    Found existing installation: requests 2.25.1
    Uninstalling requests-2.25.1:
      Successfully uninstalled requests-2.25.1
  Attempting uninstall: tensorboard-data-server
    Found existing installation: tensorboard-data-server 0.7.0
    Uninstalling tensorboard-data-server-0.7.0:
      Successfully uninstalled tensorboard-data-server-0.7.0
  Attempting uninstall: protobuf
    Found existing installation: protobuf 3.20.0
    Uninstalling protobuf-3.20.0:
      Successfully uninstalled protobuf-3.20.0
  Attempting uninstall: tqdm
    Found existing installation: tqdm 4.65.0
    Uninstalling tqdm-4.65.0:
      Successfully uninstalled tqdm-4.65.0
  Attempting uninstall: tensorboard
    Found existing installation: tensorboard 2.12.0
    Uninstalling tensorboard-2.12.0:
      Successfully uninstalled tensorboard-2.12.0
  Attempting uninstall: torch
    Found existing installation: torch 1.12.1+cu113
    Uninstalling torch-1.12.1+cu113:
      Successfully uninstalled torch-1.12.1+cu113
  Attempting uninstall: transformers
    Found existing installation: transformers 4.19.2
    Uninstalling transformers-4.19.2:
      Successfully uninstalled transformers-4.19.2
  Attempting uninstall: gitpython
    Found existing installation: GitPython 3.1.27
    Uninstalling GitPython-3.1.27:
      Successfully uninstalled GitPython-3.1.27
  Attempting uninstall: accelerate
    Found existing installation: accelerate 0.12.0
    Uninstalling accelerate-0.12.0:
      Successfully uninstalled accelerate-0.12.0
Successfully installed accelerate-0.16.0 albumentations-1.3.0 astunparse-1.6.3 bitsandbytes-0.35.4 diffusers-0.13.1 discord-webhook-1.1.0 flatbuffers-23.3.3 gast-0.4.0 gitpython-3.1.31 google-pasta-0.2.0 h5py-3.8.0 importlib-metadata-6.0.0 keras-2.11.0 libclang-15.0.6.1 lion-pytorch-0.0.7 mediapipe-0.9.1.0 modelcards-0.1.6 mypy-extensions-1.0.0 opencv-contrib-python-4.7.0.72 opencv-python-headless-4.7.0.72 opt-einsum-3.3.0 protobuf-3.19.6 pyre-extensions-0.0.23 qudida-0.0.4 requests-2.28.2 tensorboard-2.11.2 tensorboard-data-server-0.6.1 tensorflow-2.11.0 tensorflow-estimator-2.11.0 tensorflow-intel-2.11.0 tensorflow-io-gcs-filesystem-0.31.0 termcolor-2.2.0 torch-1.13.1 tqdm-4.64.1 transformers-4.26.1 typing-inspect-0.8.0 wrapt-1.15.0 xformers-0.0.17.dev464 zipp-3.15.0

[!] torch version 1.13.1 installed.
[!] torchvision version 0.13.1+cu113 installed.
[+] accelerate version 0.16.0 installed.
[+] bitsandbytes version 0.35.4 installed.
[+] diffusers version 0.13.1 installed.
[+] transformers version 4.26.1 installed.
[+] xformers version 0.0.17.dev464 installed.
Warning: caught exception 'Torch not compiled with CUDA enabled', memory monitor disabled
Launch errors detected: ['Incorrect version of torch installed.', 'Incorrect version of torchvision installed.']

#######################################################################################################
```

**Have you read the [Readme](https://github.com/d8ahazard/sd_dreambooth_extension#readme)?**
Yes.

**Have you completely restarted the stable-diffusion-webUI, not just reloaded the UI?**
Yes.

**Have you updated Dreambooth to the latest revision?**
Maybe

**Describe the bug**

(A clear and concise description of what the bug is)

As you see from the log, the Dreambooth extension uninstalls the CUDA-supported `torch 1.12.1+cu113` installed by WebUI and replaces it with CPU-only version:

See the log:
```
  Using cached torch-1.13.1-cp310-cp310-win_amd64.whl (162.6 MB)
...
  Attempting uninstall: torch
    Found existing installation: torch 1.12.1+cu113
    Uninstalling torch-1.12.1+cu113:
      Successfully uninstalled torch-1.12.1+cu113
...
Successfully installed torch-1.13.1
```
Notice that the PyTorch version that Dreambooth extension re-installs is not compiled with Cuda support.

In general it might not be the best idea to mess with the installed version of PyTorch. The SD WebUI venv is shared between all extensions. It's best when everyone plays nice.

**Provide logs**

If a crash has occurred, please provide the *entire* stack trace from the log, including the last few log messages *
before* the crash occurred.

**Environment**

What OS? Windows 10 Pro

If Windows - WSL or native? Native

What GPU are you using? RTX 3090

**Screenshots/Config**
If the issue is specific to an error while training, please provide a screenshot of training parameters or the
db_config.json file from /models/dreambooth/MODELNAME/db_config.json

**Comment by ArrowM** [comment]
Ugh, I hate python dependency management because of how difficult it is to debug. The most likely culprit is another extension. My guess for what happened is that some other extension is running a pip install with `--no-deps`, to install a lib like torchaudio or torchvision with a requirement of torch >= 0.13.0. Then when the dreambooth requirements are installed, pip re-checks everything and sees that you have a 0.12.0 version of torch and installs the newer required 0.13.1 version (but it doesn't know to include cuda). Please check your other extensions. Regardless this is almost certainly not a problem with this extension.

**Comment by ArrowM** [comment]
oh and as a fix, you can manually install torch >= 0.13.1 with cu >= 16 and you should be fine (although the recommended versions are torch==1.13.1+cu117 and torchvision==0.14.1+cu117)

**Comment by Arilziem** [comment]
I wonder if it would still be better to enforce a CUDA version or at least prompt y/n to install it unless the system is ARM64. Quite a few people run into this, ignore the 'wrong version' thing because they have no idea what it means or what to do about it and then have weird things happen (and then they either give up or come her and open issues).
